File: typescript/v1/blockytime-app/src/interfaces/trendserviceinterface.ts
import { TypeModel } from '../models/type';

export interface TrendDataPoint {
  duration: number; 
  timeLabel: string;
}

export interface TrendData {
  type: TypeModel;
  items: TrendDataPoint[];
}

export interface TrendServiceInterface {
  getTrends(startDate: Date, endDate: Date, groupBy: 'DAY' | 'MONTH'): Promise<TrendData[]>;
  getTrendsByDateString(startDateStr: string, endDateStr: string, groupBy: 'DAY' | 'MONTH'): Promise<TrendData[]>;
}

File: python/blockytime/interfaces/trendserviceinterface.py
from datetime import date
from enum import Enum
from typing import List, Protocol

from ..dtos.trenditem_dto import TrendDataDTO


class TrendGroupBy(Enum):
    DAY = "DAY"
    WEEK = "WEEK"
    MONTH = "MONTH"


class TrendServiceInterface(Protocol):

    def get_trends(
        self, start_date: date, end_date: date, group_by: TrendGroupBy
    ) -> List[TrendDataDTO]: ...
